src/test/java/com/ericsson/ei/flowtests/TestConfigs.java

```diff
@@ -0,0 +1,106 @@
+package com.ericsson.ei.flowtests;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.amqp.core.BindingBuilder;
+import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.core.TopicExchange;
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitAdmin;
+import org.springframework.util.SocketUtils;
+
+import com.mongodb.MongoClient;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+
+import de.flapdoodle.embed.mongo.distribution.Version;
+import de.flapdoodle.embed.mongo.tests.MongodForTestsFactory;
+import lombok.Getter;
+
+public class TestConfigs {
+
+    private static AMQPBrokerManager amqpBroker;
+    private static MongodForTestsFactory testsFactory;
+    private static ConnectionFactory cf;
+
+    final static Logger LOGGER = LoggerFactory.getLogger(TestConfigs.class);
+
+    @Getter
+    private static Connection conn;
+
+    @Getter
+    private static MongoClient mongoClient = null;
+
+    public static void init() throws Exception {
+        setUpMessageBus();
+        setUpEmbeddedMongo();
+    }
+
+    private static void setUpMessageBus() throws Exception {
+        LOGGER.debug("setting up message buss");
+        if (amqpBroker != null || conn != null || cf != null) {
+            return;
+        }
+
+        int port = SocketUtils.findAvailableTcpPort();
+        System.setProperty("rabbitmq.port", "" + port);
+        System.setProperty("rabbitmq.user", "guest");
+        System.setProperty("rabbitmq.password", "guest");
+        System.setProperty("waitlist.initialDelayResend", "500");
+        System.setProperty("waitlist.fixedRateResend", "100");
+        LOGGER.info("setting up message buss");
+        String config = "src/test/resources/configs/qpidConfig.json";
+        File qpidConfig = new File(config);
+        amqpBroker = new AMQPBrokerManager(qpidConfig.getAbsolutePath(), port);
+        amqpBroker.startBroker();
+        cf = new ConnectionFactory();
+        cf.setUsername("guest");
+        cf.setPassword("guest");
+
+        cf.setPort(port);
+        cf.setHandshakeTimeout(600000);
+        cf.setConnectionTimeout(600000);
+        conn = cf.newConnection();
+
```

Comment:
This method does many things, you could break it up to several

```
setUpMessageBus(){
  setSystemProperties();
  startQpidBroker();
  setupConnectionFactory();
}
```

As a simpel example

Reply:
done

Reply:
done